tests/test_qaoa_objective_sk.py

```diff
@@ -0,0 +1,187 @@
+simulators_to_run_names = get_available_simulator_names("x") + ["qiskit"]
+simulators_to_run_names_no_qiskit = get_available_simulator_names("x")
+
+
+def test_sk_obj(N=5):
+    J = get_random_J(N=N)
+
+    def sk_obj_simple(x, J):
+        obj = 0
+        for i in range(N):
+            for j in range(i, N):
+                obj += J[i, j] * (2 * x[i] - 1) * (2 * x[j] - 1)
+        return 2 * obj / np.sqrt(N)
+
+    x = rng.choice([0, 1], N)
+    assert np.isclose(sk_obj(x, J), sk_obj_simple(x, J))
+
+
+def test_parameter_normalization(N=20):
+    J = get_random_J(N=N)
+
+    f = get_qaoa_sk_objective(N, 1, J=J, parameterization="gamma beta", simulator=simulators_to_run_names[0], objective="expectation")
+    assert np.isclose(f([1], [np.pi / 8]) / N, 0.3, atol=1e-1)
+
+
+def test_variance_normalization(N=1000):
+    J = get_random_J(N=N)
+
+    terms = get_sk_terms(J)
+    terms_0 = [x[0] for x in terms]
+    var = np.var(terms_0)
+    print("var: ", var)
+
+    assert np.isclose(N * var, 1, atol=1e-2)
+
+
+def test_energy_pre_optimized():
+    N = 8
+    J = np.array(
+        [
+            [0.00000000e00, 1.15336475e00, -5.85084225e-01, 7.12352577e-01, 3.68067836e-01, -1.13088622e00, -8.64623738e-02, -3.77646427e-01],
+            [1.15336475e00, 0.00000000e00, -9.13387355e-02, -7.66210270e-01, -7.19919730e-01, -8.00833339e-01, 2.33528912e-01, -6.65389692e-01],
+            [-5.85084225e-01, -9.13387355e-02, 0.00000000e00, 2.68564121e-01, 9.91582576e-01, -9.31645738e-01, 2.33491255e-02, -7.41120517e-01],
+            [7.12352577e-01, -7.66210270e-01, 2.68564121e-01, 0.00000000e00, 9.70605475e-01, 1.02954907e00, -3.94682792e-01, -3.25847934e-04],
+            [3.68067836e-01, -7.19919730e-01, 9.91582576e-01, 9.70605475e-01, 0.00000000e00, -4.07804975e-01, 2.67471661e-01, -1.60992252e-02],
+            [-1.13088622e00, -8.00833339e-01, -9.31645738e-01, 1.02954907e00, -4.07804975e-01, 0.00000000e00, 8.64544538e-01, 7.83512686e-01],
+            [-8.64623738e-02, 2.33528912e-01, 2.33491255e-02, -3.94682792e-01, 2.67471661e-01, 8.64544538e-01, 0.00000000e00, -1.23662276e-01],
+            [-3.77646427e-01, -6.65389692e-01, -7.41120517e-01, -3.25847934e-04, -1.60992252e-02, 7.83512686e-01, -1.23662276e-01, 0.00000000e00],
+        ]
+    )
+
+    p = 8
+    gamma = np.array([0.2268, 0.4162, 0.4332, 0.4608, 0.4818, 0.5179, 0.5717, 0.6393])
+    beta = np.array([0.6151, 0.4906, 0.4244, 0.3780, 0.3224, 0.2606, 0.1884, 0.1030])
+
+    # Precomputed expected energy with the above fixed parameters: [4.207818619693583, 4.207818619693583]
+    expected_energy = len(simulators_to_run_names) * [4.207818619693583]
+
+    qaoa_objectives = [
+        get_qaoa_sk_objective(N, p, J=J, parameterization="gamma beta", simulator=simulator, objective="expectation")(gamma, beta)
+        for simulator in simulators_to_run_names
+    ]
+    assert np.allclose(qaoa_objectives, expected_energy)
+
+
+def test_sk_qaoa_convergence_with_p():
```

Comment:
Why not use `@pytest.mark.parametrize("simulator", simulators_to_run_names)` for this test instead of manually iterating over the simulators?

Reply:
Done!